yt_dlp/extractor/tumblr.py

```diff
@@ -107,116 +237,176 @@ class TumblrIE(InfoExtractor):
-                sources.append((hd_url, 'hd'))
-
-        formats = [{
-            'url': video_url,
-            'ext': 'mp4',
-            'format_id': format_id,
-            'height': int_or_none(self._search_regex(
-                r'/(\d{3,4})$', video_url, 'height', default=None)),
-            'quality': quality,
-        } for quality, (video_url, format_id) in enumerate(sources)]
-
-        self._sort_formats(formats)
+        formats = formats or [{
+            'url': media_json.get('url') or video_url,
+            'format_id': '0',
+            'width': media_json.get('width') or int_or_none(
+                self._og_search_property('video:width', webpage, default=None)),
+            'height': media_json.get('height') or int_or_none(
+                self._og_search_property('video:height', webpage, default=None)),
+        }]
 
-        # The only place where you can get a title, it's not complete,
-        # but searching in other places doesn't work for all videos
-        video_title = self._html_search_regex(
-            r'(?s)<title>(?P<title>.*?)(?: \| Tumblr)?</title>',
-            webpage, 'title')
+        uploader_id = traverse_obj(post_json, ('reblogged_root_name'), ('blog_name'))
 
         return {
             'id': video_id,
-            'title': video_title,
-            'description': self._og_search_description(webpage, default=None),
-            'thumbnail': self._og_search_thumbnail(webpage, default=None),
+            'title': (post_json.get('summary')
+                      or (blog if api_only else self._html_search_regex(
+                          r'(?s)<title>(?P<title>.*?)(?: \| Tumblr)?</title>',
+                          webpage, 'title'))),
+            # the url we're extracting from might be an original post or it might be a reblog.
+            # if it's a reblog, og:description will be the reblogger's comment, not the uploader's.
+            # content_json is the op's post, so if it exists but has no text, there's no description
+            'description': ('\n\n'.join(
+                            (
+                                item.get('text') for item in content_json
+                                if item.get('type') == 'text'
+                            ))
+                            or (None if content_json
+                                else self._og_search_description(webpage, default=None))),
+            'thumbnail': (traverse_obj(video_json, ('poster', 0, 'url'))
+                          or self._og_search_thumbnail(webpage, default=None)),
+            'uploader_id': uploader_id,
+            'uploader_url': f'https://{uploader_id}.tumblr.com/' if uploader_id else None,
             'duration': duration,
+            'like_count': post_json.get('like_count'),
+            'repost_count': post_json.get('reblog_count'),
+            'age_limit': ((18 if post_json.get('is_nsfw') else 0)
+                          if 'is_nsfw' in post_json else None),
```

Comment:
```suggestion
            'age_limit': 18 if post_json.get('is_nsfw') else 0,
```
Isn't this enough? (I'm not sure how dlp consumes age_limit tho)

Reply:
I have distinguished between 'safe' (the API explicitly told us so) and 'unknown' (we couldn't get data from the API). As far as I know this makes no practical difference, but it seems more correct and several other extractors do the same.